services/intersection-api/api/src/main/java/us/dot/its/jpo/ode/api/accessors/events/BsmMessageCountProgressionEventRepository/BsmMessageCountProgressionEventRepository.java

```diff
@@ -0,0 +1,20 @@
+package us.dot.its.jpo.ode.api.accessors.events.BsmMessageCountProgressionEventRepository;
+
+import java.util.List;
+
+import org.springframework.data.mongodb.core.query.Query;
+import us.dot.its.jpo.ode.api.models.IDCount;
+import us.dot.its.jpo.ode.api.models.DataLoader;
+import us.dot.its.jpo.conflictmonitor.monitor.models.events.BsmMessageCountProgressionEvent;
+
+public interface BsmMessageCountProgressionEventRepository extends DataLoader<BsmMessageCountProgressionEvent>{
+    Query getQuery(Integer intersectionID, Long startTime, Long endTime, boolean latest);
+
+    long getQueryResultCount(Query query);
+
+    long getQueryFullCount(Query query);
```

Comment:
suggestion: Add descriptions to these methods for better understanding. For these count methods I have to dig into the implementation to know what they are supposed to do without a description. This applies to all interfaces under events

Reply:
Created new backlog item to cover this: https://dev.azure.com/SOC-OIT/CDOT/_backlogs/backlog/CDOT%20Connected%20Vehicles/Backlog%20items/?workitem=290540